config/configauth/configauth.go

```diff
@@ -15,60 +15,48 @@
 package configauth
 
 import (
-	"context"
+	"errors"
 
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
 	"google.golang.org/grpc"
 )
 
+var (
+	errAuthenticatorNotFound    error = errors.New("authenticator not found")
+	errAuthenticatorNotProvided error = errors.New("authenticator not provided")
+)
+
 // Authentication defines the auth settings for the receiver
 type Authentication struct {
-	// The attribute (header name) to look for auth data. Optional, default value: "authentication".
-	Attribute string `mapstructure:"attribute"`
-
-	// OIDC configures this receiver to use the given OIDC provider as the backend for the authentication mechanism.
-	// Required.
-	OIDC *OIDC `mapstructure:"oidc"`
-}
-
-// OIDC defines the OpenID Connect properties for this processor
-type OIDC struct {
-	// IssuerURL is the base URL for the OIDC provider.
-	// Required.
-	IssuerURL string `mapstructure:"issuer_url"`
-
-	// Audience of the token, used during the verification.
-	// For example: "https://accounts.google.com" or "https://login.salesforce.com".
-	// Required.
-	Audience string `mapstructure:"audience"`
-
-	// The local path for the issuer CA's TLS server cert.
-	// Optional.
-	IssuerCAPath string `mapstructure:"issuer_ca_path"`
-
-	// The claim to use as the username, in case the token's 'sub' isn't the suitable source.
-	// Optional.
-	UsernameClaim string `mapstructure:"username_claim"`
-
-	// The claim that holds the subject's group membership information.
-	// Optional.
-	GroupsClaim string `mapstructure:"groups_claim"`
+	// Authenticator specifies the name of the extension to use in order to authenticate the incoming data point.
+	AuthenticatorName string `mapstructure:"authenticator"`
 }
 
-// ToServerOptions builds a set of server options ready to be used by the gRPC server
-func (a *Authentication) ToServerOptions() ([]grpc.ServerOption, error) {
-	auth, err := NewAuthenticator(*a)
-	if err != nil {
-		return nil, err
+// ToServerOption builds a set of server options ready to be used by the gRPC server
+func (a *Authentication) ToServerOption(ext map[config.NamedEntity]component.Extension) ([]grpc.ServerOption, error) {
+	if a.AuthenticatorName == "" {
+		return nil, errAuthenticatorNotProvided
 	}
 
-	// perhaps we should use a timeout here?
-	// TODO: we need a hook to call auth.Close()
-	if err := auth.Start(context.Background()); err != nil {
-		return nil, err
+	authenticator := selectAuthenticator(ext, a.AuthenticatorName)
+	if authenticator == nil {
+		return nil, errAuthenticatorNotFound
```

Comment:
Shall we wrap the error displaying the authenticator name being looking for?

Something like `fmt.Errorf("failed to resolve authenticator %q: %w", e.AuthenticatorName, err)`